### ISSUE TITLE
Experiencing SSH Issues A Few Minutes After Build Starts

### ISSUE THREAD

**Issue by nkazarian-spokeo** [hashicorp/packer#9598]
#### Overview of the Issue

We are using `amazon-ebs` builder type and we will intermittently see this error a few minutes after executing `packer build`:

```1594228095,,ui,error,==> amazon-ebs: Error waiting for SSH: Packer experienced an authentication error when trying to connect via SSH.```

#### Reproduction Steps

Run `packer build` and wait a few minutes. It is intermittent and so we have no consistent way to reproduce. It'll occur maybe 10% of the time.

### Packer version

We are running `1.5.4`

### Simplified Packer Buildfile

We let Packer generate the keypair, but we do specify `ssh_username`. We also tried adding `ssh_timeout` very recently to rule out timeout issues. This did not prove helpful. Below is a portion of our `packer.json`

```
    "builders": [
        {
            "type": "amazon-ebs",
            "region": "{{ user `region` }}",
            "communicator": "ssh",
            "ssh_pty": "true",
            "ssh_timeout": "20m",
            "source_ami": "{{ user `app_base_ami` }}",
            "instance_type": "{{ user `packer_builder_instance_type` }}",
            "ssh_username": "ec2-user",
...
```

### Operating system and Environment details

We are running on Amazon Linux 2 images

**Comment by xjables** [comment]
Could you post the debug logging for one of the runs where the issue occurs?  You can get the debug logging set by running:
 ```
PACKER_LOG=1 packer build packer.json
```

**Comment by jpatallah** [comment]
Attached is a debug log of the above issue.
[PackerError.txt](https://github.com/hashicorp/packer/files/5044660/PackerError.txt)

**Comment by xjables** [comment]
Packer is failing to authenticate ssh on an IP address that comes from a private IP range:

> Using host value: 172.31.68.174

You are most likely running the packer build in a subnet that has "auto-assign ip address" set to "no".  You can fix this by setting setting `"associate_public_ip_address": true` in your builder section.

If that doesn't work, then you are likely creating the instance in a default subnet with "auto-assign ip address" set to "no".  To fix this, you either have to set the default subnet value for "auto-assign ip address" to "yes", or select a different subnet explicitly in your builder section.

**Comment by jpatallah** [comment]
I do not believe that setting a public ip address on the packer spun instance has any bearing on this issue.  In the logs it states:
> 2020/08/08 03:43:43 packer-builder-amazon-ebs plugin: [DEBUG] handshaking with SSH
>2020/08/08 03:43:43 packer-builder-amazon-ebs plugin: [DEBUG] SSH handshake err: ssh: handshake failed: ssh: unable to authenticate, attempted methods [none publickey], no supported methods remain

The fact that it received a handshake failure means that it was able to route to the private ip and connect over port 22.  

We are utilizing the default functionality of having packer generate a temporary keypair, so not sure why it's stating:

> [DEBUG] SSH handshake err: ssh: handshake failed: ssh: unable to authenticate, attempted methods [none publickey], no supported methods remain
It should be using the packer generated key to authenticate. 

A rerun of the packer build command usually results in a successful instantiation of the instance without any configuration changes.

**Comment by xjables** [comment]
I cannot for the life of me reproduce this issue.  Are you setup on an EC2 instance within that private network and running packer on that?  Or through Jenkins or Rundeck or something?

When you have packer runs that pass, are you getting any authentication issues at all?  Or just below the default 10?  Have you tried bumping the `ssh_handshake_attempts` above the default of 10? 

Are you building on top of a security hardened image that might have different SSH config defaults?  Or do you have user data that touches SSH at all?

**Comment by jpatallah** [comment]
We have a jenkins instance ( running on EC2) set up in a private subnet we'll refer to as 'subnetP1'.  A jenkins job calls the packer binary who is configured to bring up an instance in the same subnet (subnetP1) ( packer.json attached)
[packer.json.txt](https://github.com/hashicorp/packer/files/5048427/packer.json.txt)

When we have packer runs that pass there are no authentication issues in the log.  Attached is a portion of a log where the packer run succeeded ( it was running at the same time as a different packer run that had failed)
[PackerWorking.txt](https://github.com/hashicorp/packer/files/5048442/PackerWorking.txt)

All our packer built ami's are done via automation and they all source the same amazon linux 2 ami: ami-0873b46c45c11058d . Most runs succeed, a few will fail with the ssh error and succeed on subsequent reruns. 

We haven't bumped ssh_handshake_attempts; our first pass was to bump ssh_timeout to 20m. However, it seems that isn't the necessary parameter to touch as the job fails within minutes.
